### PR TITLE
feat: poll barrier complete rpc in barrier manager worker loop

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Previously in global barrier manager, the finish of barrier_complete rpc is notified by a mpsc channel, which may receive stale rpc result after recovery. 

In this PR, we change to use a oneshot channel to notify the barrier completion of each epoch. The oneshot receiver will be pushed to a FuturesUnordered and gets polled in the global barrier manager worker loop. In this way, during failure recovery, we can avoid receiving stale completion result.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/meta/src/barrier/mod.rs [hunk 1]
```diff
@@ -1125,16 +1128,23 @@ impl GlobalBarrierManagerContext {
                     node_need_collect,
```
Comment (yezizp2012) [inline src/meta/src/barrier/mod.rs:1128]: Is it possible to manage the collect handler in rpc manager as well? For example maintain a `map<epoch, handle>` in it and remove it on complete or abort them all on recovery. It is also acceptable to remain unchanged.

**Review by yezizp2012 — APPROVED** [review]
LGTM~

**Review by BugenZhao — APPROVED** [review]
LGTM!

File: src/meta/src/barrier/mod.rs (hunk 1 above)
Comment (wenym1) [reply src/meta/src/barrier/mod.rs:1128]: Yes it's possible, but this PR is a temporary version of a final one. When we implement rpc as a bidirection stream between CN and meta, there is no need to spawn a task for rpc. So I think we can remain unchanged now.

**Comment by wenym1** [comment]
Some methods related to barrier rpc is moved to `rpc.rs` without a line of change to make the code more organized.

**Pull request merged**